core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -76,6 +76,22 @@
     return Collections.emptySet();
   }
 
+  /**
+   * Returns true if a function takes an array arguments
+   */
+  default boolean hasArrayInputs()
+  {
+    return false;
+  }
+
+  /**
+   * Returns true if function produces an array
```

Comment:
What happens if it sometimes produces an array, and sometimes doesn't, based on its input type? Similar comment to `ApplyFunction.hasArrayOutput`: is it disallowed, or should it return true/false?

Reply:
see above comment, will adjust docs to mention it shouldn't be done, but i don't have a way to really enforce it yet, need type inference i think